tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -1035,3 +1037,73 @@ def process(self, resources, event=None):
         parent_resources_ids = [p['id'] for p in parent_resources]
         parent_key = self.manager.resource_type.parent_key
         return [r for r in resources if r[parent_key] in parent_resources_ids]
+
+
+class AzureAdvisorFilter(RelatedResourceFilter):
+    """
+    Filter resources by Azure Advisor Recommendations
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: disks-with-cost-recommendations
+            resource: azure.disk
+            filters:
+              - type: advisor-recommendation
+                key: '[].properties.category'
+                value: Cost
+                value_type: swap
+                op: in
```

Comment:
nit: `op: contains` provides the same results and seems a bit clearer for docs than `op: in` + `value_type: swap`.